quickwit/quickwit-license/src/lib.rs

```diff
@@ -0,0 +1,181 @@
+use std::str::FromStr;
+use std::sync::OnceLock;
+
+use biscuit_auth::builder::Predicate;
+use biscuit_auth::builder::Rule;
+use biscuit_auth::builder::Term;
+use biscuit_auth::builder_ext::AuthorizerExt;
+use biscuit_auth::Authorizer;
+use biscuit_auth::Biscuit;
+use biscuit_auth::PublicKey;
+use tracing::error;
+use tracing::info;
+
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("invalid biscuit")]
+    Unverified,
+    #[error("invalid parameter {0}")]
+    InvalidParameter(String),
+    #[error("license expired")]
+    LicenseExpired,
+}
+
+#[cfg(test)]
+const QUICKWIT_PUBLIC_KEY_HEX: &'static str = "5a81acfc945c7c64c3e95a0c8ce2caafec7c496e76db9ff528755eca7fc16c25";
+
+#[cfg(not(test))]
+const QUICKWIT_PUBLIC_KEY_HEX: &'static str = "c8a5f551d1744547498c97638ce5c5520755cb9de0f052dcee07f90dc9663042";
+
+#[derive(Debug)]
+pub struct License {
+    pub license_level: LicenseLevel,
```

Comment:
If we want to be able to enable features separately, we would need something more here

Reply:
I think we can stick to license level, at least for the moment.

Reply:
I did not pick that from a specific license, and I am not sure of the origin of this formulation: you can google "license" "may only be used in production" to see a bunch of examples.